amgut/lib/data_access/test/test_ag_data_access.py

```diff
@@ -428,13 +434,71 @@ def test_getAGKitIDsByEmail(self):
         raise NotImplementedError()
 
     def test_ag_set_pass_change_code(self):
-        raise NotImplementedError()
+        # Generate new random code and assign it
+        testcode = ''.join(choice(ascii_letters) for i in range(10))
+        self.ag_data.ag_set_pass_change_code('REMOVED', 'tst_ULGcr', testcode)
+
+        # Acutally test the change
+        obs = self.ag_data.ag_verify_kit_password_change_code(
+            'REMOVED', 'tst_ULGcr', 'SOMELONGTHINGTHATWILLFAIL')
+        self.assertEqual(obs, False)
+        obs = self.ag_data.ag_verify_kit_password_change_code(
+            'REMOVED', 'tst_ULGcr', testcode)
+        self.assertEqual(obs, True)
+
+        # Test giving nonsense email
+        # TODO: make this raise error and test
+        self.ag_data.ag_set_pass_change_code('Fake@notarealemail.com',
+                                             'tst_ULGcr', testcode)
+
+        # Test giving bad skid
+        # TODO: make this raise error and test
+        self.ag_data.ag_set_pass_change_code('REMOVED', 'NOTINTHEDB', testcode)
 
     def test_ag_update_kit_password(self):
-        raise NotImplementedError()
+        # Generate new pass and make sure is different from current pass
+        newpass = ''.join(choice(ascii_letters) for i in range(randint(8, 15)))
+        auth = self.ag_data.authenticateWebAppUser('tst_ULGcr', newpass)
+        if auth is not False:
+            raise ValueError("Randomly generated password matches existing")
+
+        # Acutally test password change
+        self.ag_data.ag_update_kit_password('tst_ULGcr', newpass)
```

Comment:
update password actually does the update. The pass change code sets the reset code and time for the email.